src/components/forms/ProxyDeviceInput.tsx

```diff
@@ -0,0 +1,85 @@
+import { FC, useState } from "react";
+import { Input, Select } from "@canonical/react-components";
+
+interface Props {
+  onChange?: (type?: string, address?: string, port?: string) => void;
+  value?: string;
+  isConnect?: boolean;
+  isNatEnabled?: string;
+}
+
+const ProxyDeviceInput: FC<Props> = ({
+  onChange,
+  value,
+  isConnect,
+  isNatEnabled,
+}) => {
+  const [inheritedType, inheritedAddress, inheritedPort] =
+    value?.split(":") || [];
+
+  const [type, setType] = useState(inheritedType || "");
+  const [address, setAddress] = useState(inheritedAddress || "");
+  const [port, setPort] = useState(inheritedPort || "");
```

Comment:
```suggestion
  const parts = value?.split(":") || [];
  const type = parts.length > 0 ? parts[0] : "tcp";
  const address = parts.length > 1 ? parts[1] : "";
  const port = parts.length > 2 ? parts[2] : "";
```
I think we can do this without having local state in here. We can directly rely on the formik state as it comes in.